### PR TITLE
save custom ordered list start index in render JSON

### PR DESCRIPTION
**Pull request by QuietMisdreavus**
Bug/issue #, if applicable: rdar://73847907

## Summary

While cmark-gfm saves the starting index for an ordered list, swift-markdown currently discards this information. This PR is part of a set to preserve this starting index when rendering documentation.

## Dependencies

https://github.com/apple/swift-markdown/pull/22

## Testing

Add the following markdown to `Sources/SwiftDocC/SwiftDocC.docc/SwiftDocC.md`:

```markdown
2. this is a test
3. this is a test
```

Steps:
1. Build the associated swift-docc-render branch (https://github.com/apple/swift-docc-render/pull/30) and set `$DOCC_HTML_DIR` to the resulting `dist` directory.
2. `bin/preview-docs`
3. Ensure that the added list correctly starts with the index 2.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: Sources/SwiftDocC/SwiftDocC.docc/Resources/RenderNode.spec.json [hunk 1]
```diff
@@ -659,6 +659,9 @@
                             "orderedList"
                         ]
                     },
+                    "level": {
```
Comment (mportiz08) [inline Sources/SwiftDocC/SwiftDocC.docc/Resources/RenderNode.spec.json:662]: Is this meant to be `"start"` instead of `"level"`?

Comment (QuietMisdreavus) [reply Sources/SwiftDocC/SwiftDocC.docc/Resources/RenderNode.spec.json:662]: Yes! 😅 I didn't check my copy-paste all the way.

**Comment by QuietMisdreavus** [comment]
@swift-ci Please test

File: Tests/SwiftDocCTests/Rendering/RenderNodeTranslatorTests.swift [hunk 2]
```diff
@@ -102,7 +102,8 @@ class RenderNodeTranslatorTests: XCTestCase {
         }
         
         XCTAssert(discussion.content.contains(where: { block in
-            if case .orderedList(items: let items) = block,
+            if case .orderedList(start: let start, items: let items) = block,
+                start == nil,
```
Comment (franklinsch) [inline Tests/SwiftDocCTests/Rendering/RenderNodeTranslatorTests.swift:106]: Should we add a test where `start` is not `nil` and we check its value?

File: Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift [hunk 3]
```diff
@@ -45,7 +45,7 @@ public enum RenderBlockContent: Equatable {
     /// A heading with the given level.
     case heading(level: Int, text: String, anchor: String?)
     /// A list that contains ordered items.
-    case orderedList(items: [ListItem])
+    case orderedList(start: Int?, items: [ListItem])
```
Comment (franklinsch) [inline Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift:48]: Similar to the feedback I provided in Swift-Markdown, I'm wondering if this should be non-optional since all ordered lists have a start. Also, wondering if `startIndex` or `startingNumber` might be it clearer what this property refers to.

**Comment by QuietMisdreavus** [comment]
Rebased to resolve merge conflict and to bring up to date with the swift-markdown PR.

@swift-ci Please test

File: Package.resolved [hunk 4]
```diff
@@ -57,10 +57,10 @@
       },
       {
         "package": "swift-markdown",
-        "repositoryURL": "https://github.com/apple/swift-markdown.git",
+        "repositoryURL": "https://github.com/QuietMisdreavus/swift-markdown.git",
         "state": {
-          "branch": "main",
-          "revision": "69f31f83a5dca9a3401d3e8fc8ff74804b0dd0ca",
+          "branch": "list-numbering",
+          "revision": "0cdc7269aa395115e50b25f388ffd184a4f04787",
```
Comment (franklinsch) [inline Package.resolved:63]: Small reminder to un-stage this before merging

**Review by franklinsch — APPROVED** [review]
Looks great, thanks!

**Comment by QuietMisdreavus** [comment]
@swift-ci Please test
